hashring/hashring.go

```diff
@@ -76,6 +80,7 @@ func New(hashfunc func([]byte) uint32, replicaPoints int) *HashRing {
 		hashfunc: func(str string) int {
 			return int(hashfunc([]byte(str)))
 		},
+		logger: logging.Logger("ring"),
```

Comment:
I don't think the train has left the station (nothing is impossible :wink:)  - I don't even think this is difficult to achieve.

The root-level logger just needs to have a `WithFields` implementation with a way to set the value of the `local` field from outside. The identity resolver can then just set it as soon as it knows it.

Anyway, outside the scope of this PR....